netsnmpagent.py

```diff
@@ -400,12 +399,26 @@ def _prepareRegistration(self, oidstr, writable = True):
 		else:
 			# Interpret the given oidstr as the oid itself.
 			try:
-				parts = [c_oid(long(x) if sys.version_info <= (3,) else int(x)) for x in oidstr.split('.')]
+				parts = [c_oid(long(x) if sys.version_info <= (3,) else int(x)) for x in oidstr.strip('.').split('.')]
 			except ValueError:
 				raise netsnmpAgentException("Invalid OID (not using MIB): {0}".format(oidstr))
 
 			oid = (c_oid * len(parts))(*parts)
 			oid_len = ctypes.c_size_t(len(parts))
+		return (oid, oid_len)
+
+	def _prepareRegistration(self, oidstr, writable = True):
+		""" Prepares the registration of an SNMP object.
+
+		    "oidstr" is the OID to register the object at.
+		    "writable" indicates whether "snmpset" is allowed. """
+
+		# Make sure the agent has not been start()ed yet
+		if self._status != netsnmpAgentStatus.REGISTRATION:
+			raise netsnmpAgentException("Attempt to register SNMP object " \
```

Comment:
I think the trailing `\` here is unnecessary since it's within parentheses

Reply:
Will remove